### PR TITLE
Change pmb site pairing

### PR DESCRIPTION
**Pull request by Seeker1911**
Pair at biospecimen collected site.

### CODE REVIEW

File: rest-api/test/unit_test/offline_test/biobank_samples_pipeline_mysql_test.py [hunk 1]
```diff
@@ -90,6 +90,7 @@ def _insert_order(self, participant, order_id, tests, order_time, finalized_test
         participantId=participant.participantId,
         sourceSiteId=1,
         finalizedSiteId=1,
+        collectedSiteId=1,
```
Comment (danqrodney) [inline rest-api/test/unit_test/offline_test/biobank_samples_pipeline_mysql_test.py]: You may want to clear out finalizedSiteId here, or set it to something else, to prove collected site ID is being used now.

We should add a check here:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/dao/biobank_order_dao.py#L246

To ensure that all biobank orders created henceforth have collected site info. (Which appears to be the case now, but wasn't the case back in June of last year.)

File: rest-api/test/unit_test/dao_test/physical_measurements_dao_test.py [hunk 2]
```diff
@@ -93,6 +94,8 @@ def testInsert_rightParticipantId(self):
     summary = ParticipantSummaryDao().get(self.participant.participantId)
     self.assertEquals(PhysicalMeasurementsStatus.COMPLETED, summary.physicalMeasurementsStatus)
     self.assertEquals(TIME_2, summary.physicalMeasurementsTime)
+    self.assertEquals(summary.biospecimenFinalizedSiteId, None)
```
Comment (danqrodney) [inline rest-api/test/unit_test/dao_test/physical_measurements_dao_test.py]: Same as above... there's no reason to think this field would have been set.

What we need to do is add a test that checks that we don't re-pair the participant based on physical measurements once a biobank order has been received and this field *is* set.

File: rest-api/test/unit_test/api_test/biobank_order_api_test.py [hunk 3]
```diff
@@ -65,8 +65,12 @@ def test_auto_pair_updates_participant_and_summary(self):
     p_paired = self.participant_dao.get(self.participant.participantId)
     self.assertNotEqual(p_paired.hpoId, UNSET_HPO_ID)
     self.assertIsNotNone(p_paired.providerLink)
+
     s_paired = self.summary_dao.get(self.participant.participantId)
+
     self.assertNotEqual(s_paired.hpoId, UNSET_HPO_ID)
+    self.assertEqual(s_paired.biospecimenFinalizedSiteId, s_paired.siteId)
+    self.assertEqual(s_paired.physicalMeasurementsFinalizedSiteId, None)
```
Comment (danqrodney) [inline rest-api/test/unit_test/api_test/biobank_order_api_test.py]: I don't understand this test... there's no reason to think this field would have been set.

File: rest-api/test/unit_test/api_test/biobank_order_api_test.py [hunk 4]
```diff
@@ -65,8 +65,12 @@ def test_auto_pair_updates_participant_and_summary(self):
     p_paired = self.participant_dao.get(self.participant.participantId)
     self.assertNotEqual(p_paired.hpoId, UNSET_HPO_ID)
     self.assertIsNotNone(p_paired.providerLink)
+
     s_paired = self.summary_dao.get(self.participant.participantId)
+
     self.assertNotEqual(s_paired.hpoId, UNSET_HPO_ID)
+    self.assertEqual(s_paired.biospecimenFinalizedSiteId, s_paired.siteId)
```
Comment (danqrodney) [inline rest-api/test/unit_test/api_test/biobank_order_api_test.py]: So we should make sure this test has different site ID for collected vs. finalized, and that it picks the collected site.

Comment (Seeker1911) [reply rest-api/test/unit_test/api_test/biobank_order_api_test.py]: done

File: rest-api/test/unit_test/api_test/biobank_order_api_test.py (hunk 3 above)
Comment (Seeker1911) [reply rest-api/test/unit_test/api_test/biobank_order_api_test.py]: removed.

File: rest-api/test/unit_test/dao_test/physical_measurements_dao_test.py (hunk 2 above)
Comment (Seeker1911) [reply rest-api/test/unit_test/dao_test/physical_measurements_dao_test.py]: new test added

File: rest-api/test/unit_test/offline_test/biobank_samples_pipeline_mysql_test.py (hunk 1 above)
Comment (Seeker1911) [reply rest-api/test/unit_test/offline_test/biobank_samples_pipeline_mysql_test.py]: set to 2

File: rest-api/test/unit_test/api_test/biobank_order_api_test.py [hunk 5]
```diff
@@ -69,9 +70,41 @@ def test_auto_pair_updates_participant_and_summary(self):
     s_paired = self.summary_dao.get(self.participant.participantId)
 
     self.assertNotEqual(s_paired.hpoId, UNSET_HPO_ID)
-    self.assertEqual(s_paired.biospecimenFinalizedSiteId, s_paired.siteId)
-    self.assertEqual(s_paired.physicalMeasurementsFinalizedSiteId, None)
+    self.assertEqual(s_paired.biospecimenCollectedSiteId, s_paired.siteId)
+    self.assertNotEqual(s_paired.biospecimenCollectedSiteId, s_paired.biospecimenFinalizedSiteId)
+    # print s_paired.biospecimenCollectedSiteId, '...collected id'
```
Comment (danqrodney) [inline rest-api/test/unit_test/api_test/biobank_order_api_test.py]: Remove these?

File: rest-api/test/unit_test/api_test/biobank_order_api_test.py [hunk 6]
```diff
@@ -69,9 +70,41 @@ def test_auto_pair_updates_participant_and_summary(self):
...
+    # print s_paired.participantId
+    # print self.participant.participantId
+    # print self.participant.participantId
+    # self.send_consent(self.participant.participantId)
+    # # self._insert_measurements(datetime.datetime.utcnow().isoformat())
+    # #
+    # # print s_paired.biospecimenCollectedSiteId, '...collected id'
+    self.assertEqual(s_paired.biospecimenCollectedSiteId, s_paired.siteId)
```
Comment (danqrodney) [inline rest-api/test/unit_test/api_test/biobank_order_api_test.py]: This looks like a repeat of line 73.

File: rest-api/test/unit_test/dao_test/physical_measurements_dao_test.py (hunk 2 above)
Comment (danqrodney) [reply rest-api/test/unit_test/dao_test/physical_measurements_dao_test.py]: OK I think you can remove this line if you want.

File: rest-api/test/unit_test/api_test/biobank_order_api_test.py (hunk 5 above)
Comment (Seeker1911) [reply rest-api/test/unit_test/api_test/biobank_order_api_test.py]: just testing you

File: rest-api/test/unit_test/api_test/biobank_order_api_test.py (hunk 6 above)
Comment (Seeker1911) [reply rest-api/test/unit_test/api_test/biobank_order_api_test.py]: I just wanted to be REALLY sure

File: rest-api/test/unit_test/dao_test/physical_measurements_dao_test.py (hunk 2 above)
Comment (Seeker1911) [reply rest-api/test/unit_test/dao_test/physical_measurements_dao_test.py]: done

File: rest-api/test/unit_test/api_test/biobank_order_api_test.py (hunk 6 above)
Comment (danqrodney) [reply rest-api/test/unit_test/api_test/biobank_order_api_test.py]: LOL

**Pull request merged**